MultiServer.py

```diff
@@ -727,6 +749,15 @@ def notify_hints(self, team: int, hints: typing.List[NetUtils.Hint], only_new: b
                 for client in clients:
                     async_start(self.send_msgs(client, client_hints))
 
+    def get_hint(self, team: int, finding_player: int, seeked_location: int) -> typing.Optional[Hint]:
+        for hint in self.hints[team, finding_player]:
+            if hint.location == seeked_location:
+                return hint
+        return None
+    
+    def replace_hint(self, team: int, slot: int, old_hint: Hint, new_hint: Hint) -> None:
+        self.hints[team, slot] = {new_hint if hint == old_hint else hint for hint in self.hints[team, slot]}
```

Comment:
The whole point of a set is that lookup, and by extension remove-by-element, are super fast

Just do:

```
self.hints.remove(old_hint)
self.hints.add(hint)
```

Or, if the safety is necessary:

```
if hint in self.hints[team, slot]:
    self.hints.remove(old_hint)
    self.hints.add(hint)
```

That latter option is fully equivalent & both are leagues faster than this comprehension.